framework/src/modules/chain/synchronizer/synchronizer.js

```diff
@@ -164,7 +164,7 @@ class Synchronizer {
 		for (const mechanism of this.mechanisms) {
 			if (await mechanism.isValidFor(receivedBlock)) {
 				if (selectedMechanism) {
-					throw new Error('Mechanisms are conflicting'); // TODO: Add better message
+					throw new Error('Two mechanisms cannot be valid at the same time');
```

Comment:
`Two` can be misleading, as one can register more than 2 aswell. it would be better if you could make it `multiple`? that way it's generic.

Reply:
Ok will change it :)